### PR TITLE
NEPT-1239: Merge the europa-atomium feature branch into the master

### PR DESCRIPTION
**Pull request by voidtek**
### Description

Merge the europa-atomium feature branch into the master

### Change log

- Added: EC Europa feature.
- Added: NextEuropa Formatters modules that allows to use ECL components in Views and Fields.
- Changed: Externalize Ec_resp theme. 
- Changed: New logic to run the Behat tests based behat.

### Commands

```sh
drush cc all

```

### CODE REVIEW

File: profiles/common/modules/custom/nexteuropa_ecl_migrate/README.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+This module includes:
+- Footer DT menus
+- Europa main menu
+- Europa 'utility' region (previously 'tools' region)
+
+Notes
+-----
+
+For implementation details of ECL refer to:
+- https://github.com/ec-europa/ec-europa-theme
```
Comment (jfhovinne) [inline profiles/common/modules/custom/nexteuropa_ecl_migrate/README.md]: To be updated to point to new repo

File: profiles/common/modules/custom/nexteuropa_ecl_migrate/nexteuropa_ecl_migrate.info [hunk 2]
```diff
@@ -0,0 +1,4 @@
+name = NextEuropa ECL Migrate
+description = Migrate blocks of content from ec_resp to Europa.
```
Comment (jfhovinne) [inline profiles/common/modules/custom/nexteuropa_ecl_migrate/nexteuropa_ecl_migrate.info]: Should be 'from ec_resp to ec_europa'

File: profiles/common/modules/custom/nexteuropa_ecl_migrate/nexteuropa_ecl_migrate.module [hunk 3]
```diff
@@ -0,0 +1,220 @@
+<?php
+
+/**
+ * @file
+ * Code to migrate blocks from ec_resp to europa theme.
```
Comment (jfhovinne) [inline profiles/common/modules/custom/nexteuropa_ecl_migrate/nexteuropa_ecl_migrate.module]: ec_europa

File: profiles/common/modules/features/multisite_readspeaker/js/ReadSpeaker/ReadSpeaker.pub.Config.js [hunk 4]
```diff
@@ -1,7 +1,7 @@
...
...ra.",hlscrollinfo:decodeURIComponent("Activa el despla%C3%A7ament vertical autom%C3%A0tic. Activat, la p%C3%A0gina es despla%C3%A7a verticalment de manera autom%C3%A0tica per seguir a l'efecte ressaltat del text."),hliconinfo:"Controlar si el minireproductor es mostrar� quan es selecciona text.",close:"Tancar",sliderseek:decodeURIComponent("Control de temps de reproducci%C3%B3"),slidervolume:"Control de volum",slidervolumedesc:"Canviar volum",touheading:decodeURIComponent("Condicions d'%C3%BAs per a la desc�rrega d'arxius d'%C3%A0udio."),toubody:decodeURIComponent("%3Cp%3EGr%C3%A0cies%20per%20utilitzar%20el%20servei%20Readspeaker.%3C%2Fp%3E%3Cp%3ENom%C3%A9s%20es%20permet%20utilitzar%20un%20fitxer%20d'%C3%A0udio%20que%20va%20a%20descarregar%20per%20a%20%C3%BAs%20personal.%20Aix%C3%B2%20vol%20dir%20que%20no%20est%C3%A0%20permesa%20la%20distribuci%C3%B3%20d'aquest%20arxiu%20d'%C3%A0udio%20amb%20altres%20persones%2C%20o%20el%20seu%20%C3%BAs%20per%20a%20qualsevol%20altre%20prop%C3%B2sit%20diferent%20a%20escoltar%20vost%C3%A8%20personalment.%3C%2Fp%3E"),touaccept:decodeURIComponent("Accepta%20i%20baixa%20l'arxiu%20d'%C3%A0udio"),toudecline:decodeURIComponent("cancel%20%E2%80%A2%20lar")},
```
Comment (jfhovinne) [inline profiles/common/modules/features/multisite_readspeaker/js/ReadSpeaker/ReadSpeaker.pub.Config.js]: Why was it updated within europa-atomium ?

File: profiles/common/modules/features/multisite_readspeaker/js/ReadSpeaker/ReadSpeaker.pub.Config.js [hunk 5]
```diff
@@ -17,9 +17,9 @@ hlwordinfo:decodeURIComponent("Ustaw%20styl%20koloru%20do%20pod%C5%9Bwietlanych%
...
...simult%C3%A2neo com a sua leitura."),hloptionsinfo:decodeURIComponent("Controla o comportamento do destaque simult%C3%A2neo."),hlwordinfo:"Define o estilo ou a cor das palavras em destaque.",hlsentinfo:"Define o estilo ou a cor das frases em destaque.",hlspeedinfo:"Controla a velocidade de leitura.",hlscrollinfo:decodeURIComponent("Activa o scroll autom%C3%A1tico. Quando ligado, a p%C3%A1gina entra em scroll autom�tico para acompanhar o destaque do texto."),hliconinfo:decodeURIComponent("Controla se deve ou n%C3%A3o surgir um mini player quando %C5%BD seleccionado texto."),close:"Fechar",sliderseek:"Barra de busca",slidervolume:"Barra de volume de som",slidervolumedesc:"Alterar volume de som",touheading:decodeURIComponent("Termos de uso para o %C3%A1udio descarregado"),toubody:decodeURIComponent("<p>Obrigado por usar o servi%C3%A7o ReadSpeaker.</p><p>O ficheiro de %C3%A1udio que est%C3%A1 prestes a descarregar %C3%A9 apenas para o seu uso pessoal. Isto significa que n%C3%A3o lhe %C3%A9 permitido distribuir o ficheiro por outras pessoas, ou us%C3%A1-lo para outro prop%C3%B3sito que n%C3%A3o seja o de o ouvir.</p>"),touaccept:"Aceitar e Descarregar o ficheiro",toudecline:"Cancelar"},
```
Comment (jfhovinne) [inline profiles/common/modules/features/multisite_readspeaker/js/ReadSpeaker/ReadSpeaker.pub.Config.js]: Why was it updated within europa-atomium ?

File: profiles/common/modules/features/ecas_env/ecas_env.module [hunk 6]
```diff
@@ -20,7 +20,7 @@ function ecas_env_menu_link_alter(&$item) {
     if (!isset($item['customized']) || (isset($item['customized']) && $item['customized'] == 0)) {
       switch ($item['link_path']) {
         case 'ecas':
-          $item['options']['attributes']['title'] = t('Log in');
+          $item['options']['attributes']['title'] = t('EU login');
```
Comment (jfhovinne) [inline profiles/common/modules/features/ecas_env/ecas_env.module]: Why was it updated within europa-atomium ?

File: profiles/common/modules/features/multisite_readspeaker/js/ReadSpeaker/ReadSpeaker.pub.Config.js [hunk 7]
```diff
@@ -17,9 +17,9 @@ hlwordinfo:decodeURIComponent("Ustaw%20styl%20koloru%20do%20pod%C5%9Bwietlanych%
...
...n[l[m]]={};n=n[l[m]]}}else{m=g}n[l[m]]=k;return k}}},a=function(i,g){g=g||e;for(var h in i){if(i.hasOwnProperty(h)){if(g[h]!==undefined){if($rs.isArray(g[h])){g[h]=i[h]}else{if(typeof i[h]=="object"&&Object.size(i[h])>0&&!$rs.isArray(i[h])){a(i[h],g[h])}else{g[h]=i[h]}}}else{g[h]=i[h]}}}};return{init:function(){rspkr.log("[rspkr.cfg] Initialized!")},item:function(i,h,g){if(h===undefined){return d(i,g)}else{return f(i,h)}},getPhrase:function(h,i,j){j=j||"en_us";i=i||rspkr.c.data.params.lang||j;var g;if(g=this.item("phrases."+i+"."+h)){return g}else{if(g=this.item("phrases."+j+"."+h)){return g}else{return undefined}}},getPhrases:function(h,i){i=i||"en_us";var g;if(g=this.item("phrases."+h)){return g}else{if(g=this.item("phrases."+i)){return g}else{return undefined}}},getSettingsConf:function(){return c},isSentOnly:function(l){var k=this.item("general.sentOnlyLang")||[];for(var h=0,g=k.length;h<g;h++){if(k[h]==l){return true}}return false},setup:function(){if(typeof window.rsConf=="object"){a(window.rsConf)}rspkr.devt("onConfigSetup",window)},execCallback:function(i,h,g){if(typeof this.item(i,undefined,true)=="function"){this.item(i,undefined,true).apply(h,$rs.isArray(g)?g:[])}}}})();
```
Comment (jfhovinne) [inline profiles/common/modules/features/multisite_readspeaker/js/ReadSpeaker/ReadSpeaker.pub.Config.js]: Why was it updated within europa-atomium ?

File: profiles/common/modules/features/multisite_settings/multisite_settings_core/multisite_settings_core.strongarm.inc [hunk 8]
```diff
@@ -56,6 +56,13 @@ function multisite_settings_core_strongarm() {
   $strongarm = new stdClass();
   $strongarm->disabled = FALSE; /* Edit this to true to make a default strongarm disabled initially */
   $strongarm->api_version = 1;
+  $strongarm->name = 'theme_default';
+  $strongarm->value = 'ec_europa';
```
Comment (jfhovinne) [inline profiles/common/modules/features/multisite_settings/multisite_settings_core/multisite_settings_core.strongarm.inc]: Is this on purpose ?

File: profiles/multisite_drupal_standard/modules/custom/nexteuropa_core_views [hunk 9]
```diff
@@ -0,0 +1 @@
+../../../common/modules/custom/nexteuropa_core_views
```
Comment (jfhovinne) [inline profiles/multisite_drupal_standard/modules/custom/nexteuropa_core_views]: Should be removed ?

File: tests/features/change_tracking_workflow.feature [hunk 10]
```diff
@@ -156,6 +156,8 @@ Feature: Change tracking features
       | blocked                                                                                                                                                                                                                        |
       | <p>Article body<span class=\"ice-ins ice-cts-1\" data-changedata=\"\" data-cid=\"2\" data-last-change-time=\"1471619239866\" data-time=\"1471619234543\" data-userid=\"1\" data-username=\"admin\"> additional content</span></p> |
 
+  @theme_wip
+  # Failed with the EUROPA theme because of the bug covered by the ticket NEPT-1213.
```
Comment (jfhovinne) [inline tests/features/change_tracking_workflow.feature]: Should be 'EC Europa theme'

File: tests/features/change_tracking_workflow.feature [hunk 11]
```diff
@@ -200,6 +202,9 @@ Feature: Change tracking features
       | <p>No ice-ins or ice-del tracking change <a href=\"http://www.europa.eu/newsroom\">The latest news</a></p> | <p>No ice-ins or ice-del tracking change <a href=\"http://www.europa.eu/newsroom\">The latest news</a></p> |
       | <p>No tracking change <a href=\"http://www.europa.eu/newsroom\">The latest news</a></p>                    | <p>No tracking change <a href=\"http://www.europa.eu/newsroom\">The latest news</a></p>                    |
 
+  @theme_wip
+  # It is in wip for the europa theme because it implies a step referring a
```
Comment (jfhovinne) [inline tests/features/change_tracking_workflow.feature]: 'EC Europa theme'

File: tests/features/contact_form.feature [hunk 12]
```diff
@@ -4,7 +4,8 @@ Feature: Contact Form
   As a citizen of the European Union
   I want to be able to access contact forms
 
-  @javascript
+  @javascript @theme_wip
+  # Failed with the EUROPA theme because of the bug covered by the ticket NEPT-1216.
```
Comment (jfhovinne) [inline tests/features/contact_form.feature]: 'EC Europa theme'

File: tests/features/contact_form.feature [hunk 13]
```diff
@@ -21,7 +22,8 @@ Feature: Contact Form
       | error messages                   |
       | Math question field is required. |
 
-  @javascript
+  @javascript @theme_wip
+  # Failed with the EUROPA theme because of the bug covered by the ticket NEPT-1216.
```
Comment (jfhovinne) [inline tests/features/contact_form.feature]: 'EC Europa theme'

File: tests/features/content_level_language_switcher.feature [hunk 14]
```diff
@@ -27,14 +27,17 @@ Feature: Content level language switcher tests
...
+    Then I should not see an "nept_element:block:page-language-switcher" element
     When I go to "content/title-english_de"
-    Then I should not see an ".block-language-selector-page" element
+    Then I should not see an "nept_element:block:page-language-switcher" element
     When I go to "content/title-english_it"
-    Then I should see an ".block-language-selector-page" element
+    Then I should see an "nept_element:block:page-language-switcher" element
 
+  @theme_wip
+  # It is in wip because of the page languag switcher that is not adapted in the
```
Comment (jfhovinne) [inline tests/features/content_level_language_switcher.feature]: language

File: tests/features/content_level_language_switcher.feature [hunk 15]
```diff
@@ -27,14 +27,17 @@ Feature: Content level language switcher tests
...
+  # europa theme.
```
Comment (jfhovinne) [inline tests/features/content_level_language_switcher.feature]: EC Europa theme

File: tests/features/drupal_mail.feature [hunk 16]
```diff
@@ -10,6 +10,9 @@ Feature: Drupal Mail features
       | name  | mail         | roles        |
       | foo   | foo@bar.com  | contributor  |
 
+  @theme_wip
+  # It is in wip for the europa theme because it implies a step referring a
```
Comment (jfhovinne) [inline tests/features/drupal_mail.feature]: EC Europa theme

File: tests/features/ecas.feature [hunk 17]
```diff
@@ -16,12 +16,18 @@ Scenario: Administrators can add the last update block in a region
   When I visit "admin/structure/block"
   Then I should see the text "EU Login"
 
+@theme_wip
+# It is in wip for the europa theme because it implies a step referring a
```
Comment (jfhovinne) [inline tests/features/ecas.feature]: EC Europa theme

File: tests/features/ecas.feature [hunk 18]
```diff
@@ -16,12 +16,18 @@ Scenario: Administrators can add the last update block in a region
...
+# region. This must be evaluate deeper before being able to know how to deal with.
 Scenario: Logged in users can logout with the EU Login block
   Given I am logged in as a user with the "administrator" role
   And that the block "ecas" from module "ecas" is assigned to the region "sidebar_right"
   When I am on the homepage
   Then I should see the link "Logout" in the "sidebar_right" region
 
+@theme_wip
+# It is in wip for the europa theme because it implies a step referring a
```
Comment (jfhovinne) [inline tests/features/ecas.feature]: EC Europa theme

File: tests/features/ecas.feature [hunk 19]
```diff
@@ -16,12 +16,18 @@ Scenario: Administrators can add the last update block in a region
   When I visit "admin/structure/block"
   Then I should see the text "EU Login"
 
+@theme_wip
+# It is in wip for the europa theme because it implies a step referring a
+# region. This must be evaluate deeper before being able to know how to deal with.
```
Comment (jfhovinne) [inline tests/features/ecas.feature]: evaluated

File: tests/features/ecas.feature [hunk 20]
```diff
@@ -16,12 +16,18 @@ Scenario: Administrators can add the last update block in a region
...
 Scenario: Logged in users can logout with the EU Login block
   Given I am logged in as a user with the "administrator" role
   And that the block "ecas" from module "ecas" is assigned to the region "sidebar_right"
   When I am on the homepage
   Then I should see the link "Logout" in the "sidebar_right" region
 
+@theme_wip
+# It is in wip for the europa theme because it implies a step referring a
+# region. This must be evaluate deeper before being able to know how to deal with.
```
Comment (jfhovinne) [inline tests/features/ecas.feature]: evaluated

File: tests/features/images.feature [hunk 21]
```diff
@@ -4,6 +4,9 @@ Feature: Images
   As a contributor
   I can attach images to content
 
+  @theme_wip
+  # It is in WIP because the image display is not integrated in europa yet.
```
Comment (jfhovinne) [inline tests/features/images.feature]: EC Europa

File: tests/features/language_negociation.feature [hunk 22]
```diff
@@ -28,6 +28,9 @@ Feature: Second favorite language tests
     | content/title-english_bg | ?2nd-language=pt-pt | Este titulo e Portugues  |
     | content/title-english_bg | ?2nd-language=de    | This title is in English |
 
+  @theme_wip
+  # It is in wip for the europa theme because it implies a step referring a
```
Comment (jfhovinne) [inline tests/features/language_negociation.feature]: EC Europa theme

File: tests/features/language_negociation.feature [hunk 23]
```diff
@@ -28,6 +28,9 @@ Feature: Second favorite language tests
...
+  # region. This must be evaluate deeper before being able to know how to deal with.
```
Comment (jfhovinne) [inline tests/features/language_negociation.feature]: evaluated

File: tests/features/last_update.feature [hunk 24]
```diff
@@ -14,20 +14,26 @@ Scenario: Administrators can add the last update block in a region
   When I visit "admin/structure/block"
   Then I should see the text "Last update date"
 
+@RevertBlockConfiguration @theme_wip
+# It is in wip for the europa theme because it implies a step referring a
```
Comment (jfhovinne) [inline tests/features/last_update.feature]: EC Europa theme

File: tests/features/last_update.feature [hunk 25]
```diff
@@ -14,20 +14,26 @@ Scenario: Administrators can add the last update block in a region
...
+# region. This must be evaluate deeper before being able to know how to deal with.
```
Comment (jfhovinne) [inline tests/features/last_update.feature]: evaluated

File: tests/features/last_update.feature [hunk 26]
```diff
@@ -14,20 +14,26 @@ Scenario: Administrators can add the last update block in a region
...
 Scenario: Check that the last update block is not shown if a node is not published
   Given that the block "last_update" from module "nexteuropa_lastupdate" is assigned to the region "footer"
   When I go to "node/add/page"
   And I fill in "Title" with "Page title"
   And I press "Save"
-  Then I should not see an ".last-update" element
+  Then I should not see an "nept_element:block:last-update" element
 
+@RevertBlockConfiguration @theme_wip
+# It is in wip for the europa theme because it implies a step referring a
```
Comment (jfhovinne) [inline tests/features/last_update.feature]: EC Europa theme

File: tests/features/last_update.feature [hunk 27]
```diff
@@ -14,20 +14,26 @@ Scenario: Administrators can add the last update block in a region
...
+# region. This must be evaluate deeper before being able to know how to deal with.
```
Comment (jfhovinne) [inline tests/features/last_update.feature]: evaluated

File: tests/features/multilingual.feature [hunk 28]
```diff
@@ -12,6 +12,9 @@ Feature: Multilingual features
       | de        |
       | it        |
 
+  @theme_wip
+  # It is in wip for the europa theme because it implies a step referring a
```
Comment (jfhovinne) [inline tests/features/multilingual.feature]: ...

**Comment by drupol** [comment]
:+1:

File: tests/features/images.feature (hunk 21 above)
Comment (gillesdeudon) [inline tests/features/images.feature]: It is not a problem. this comment will be disappear once the wip is fixed

File: tests/features/language_negociation.feature (hunk 22 above)
Comment (gillesdeudon) [inline tests/features/language_negociation.feature]: It is not a problem. this comment will be disappear once the wip is fixed

File: tests/features/language_negociation.feature (hunk 23 above)
Comment (gillesdeudon) [inline tests/features/language_negociation.feature]: It is not a problem. this comment will be disappear once the wip is fixed

File: tests/features/last_update.feature (hunk 24 above)
Comment (gillesdeudon) [inline tests/features/last_update.feature]: It is not a problem. this comment will be disappear once the wip is fixed

File: tests/features/last_update.feature (hunk 25 above)
Comment (gillesdeudon) [inline tests/features/last_update.feature]: It is not a problem. this comment will be disappear once the wip is fixed

File: tests/features/last_update.feature (hunk 26 above)
Comment (gillesdeudon) [inline tests/features/last_update.feature]: It is not a problem. this comment will be disappear once the wip is fixed

File: tests/features/last_update.feature (hunk 27 above)
Comment (gillesdeudon) [inline tests/features/last_update.feature]: It is not a problem. this comment will be disappear once the wip is fixed

File: tests/features/multilingual.feature (hunk 28 above)
Comment (gillesdeudon) [inline tests/features/multilingual.feature]: It is not a problem ...

File: tests/features/ecas.feature (hunk 20 above)
Comment (gillesdeudon) [inline tests/features/ecas.feature]: It is not a problem. this comment will be disappear once the wip is fixed

File: tests/features/ecas.feature (hunk 19 above)
Comment (gillesdeudon) [inline tests/features/ecas.feature]: It is not a problem. this comment will be disappear once the wip is fixed

File: tests/features/ecas.feature (hunk 18 above)
Comment (gillesdeudon) [inline tests/features/ecas.feature]: It is not a problem. this comment will be disappear once the wip is fixed

File: tests/features/ecas.feature (hunk 17 above)
Comment (gillesdeudon) [inline tests/features/ecas.feature]: It is not a problem. this comment will be disappear once the wip is fixed

File: tests/features/drupal_mail.feature (hunk 16 above)
Comment (gillesdeudon) [inline tests/features/drupal_mail.feature]: It is not a problem. this comment will be disappear once the wip is fixed

File: tests/features/content_level_language_switcher.feature (hunk 15 above)
Comment (gillesdeudon) [inline tests/features/content_level_language_switcher.feature]: It is not a problem. this comment will be disappear once the wip is fixed

File: tests/features/content_level_language_switcher.feature (hunk 14 above)
Comment (gillesdeudon) [inline tests/features/content_level_language_switcher.feature]: It is not a problem. this comment will be disappear once the wip is fixed

File: tests/features/contact_form.feature (hunk 13 above)
Comment (gillesdeudon) [inline tests/features/contact_form.feature]: It is not a problem. this comment will be disappear once the wip is fixed

File: tests/features/contact_form.feature (hunk 12 above)
Comment (gillesdeudon) [inline tests/features/contact_form.feature]: It is not a problem. this comment will be disappear once the wip is fixed

File: tests/features/change_tracking_workflow.feature (hunk 11 above)
Comment (gillesdeudon) [inline tests/features/change_tracking_workflow.feature]: It is not a problem. this comment will be disappear once the wip is fixed

File: tests/features/change_tracking_workflow.feature (hunk 10 above)
Comment (gillesdeudon) [inline tests/features/change_tracking_workflow.feature]: It is not a problem. this comment will be disappear once the wip is fixed

File: profiles/common/modules/features/ecas_env/ecas_env.module (hunk 6 above)
Comment (gillesdeudon) [inline profiles/common/modules/features/ecas_env/ecas_env.module]: This change comes to remark about the fact that Log in is confusing for site where EU Login is not used for contributors.

**Comment by jfhovinne** [comment]
:-1:

File: tests/features/change_tracking_workflow.feature (hunk 10 above)
Comment (jfhovinne) [inline tests/features/change_tracking_workflow.feature]: It is incorrect, but not a blocker for this merge (idem for same typos below).

**Comment by drupol** [comment]
👍

**Comment by netlooker** [comment]
:+1:

**Comment by jfhovinne** [comment]
:+1:

**Pull request merged**